### PR TITLE
Add a basic interactive interface

### LINKED ISSUE #40: Basic shiny app
Using https://github.com/jcheng5/shinychat

### PR DESCRIPTION
**Pull request by hadley**
Fixes #40

### CODE REVIEW

File: R/shiny.R [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#'
+#' @param chat A chat object created by [new_chat_openai()] or friends.
+#' @export
+#' @returns A [shiny::shinyApp].
+chatApp <- function(chat) {
+  check_installed(c("bslib", "shiny", "shinychat"))
+
+  ui <- bslib::page_fluid(
+    shinychat::chat_ui("chat"),
+    shiny::actionButton(
```
Comment (hadley) [inline R/shiny.R:13]: Not sure if this is the best way to add a close button but it's what claude suggested 😄

**Comment by hadley** [comment]
I wonder if `$console()` should instead be a function, or if this should instead be a method?

File: R/shiny.R [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#' Open an interactive chat application
+#'
+#' Use `chatApp()` to launch an interactive shiny app for any chat instance.
+#'
+#' @param chat A chat object created by [new_chat_openai()] or friends.
+#' @export
+#' @returns A [shiny::shinyApp].
+chatApp <- function(chat) {
```
Comment (jcheng5) [inline R/shiny.R:8]: Not `chat_app`? 🤔

Comment (hadley) [reply R/shiny.R:8]: I was trying to be consistent with shiny 😆

File: R/shiny.R [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      stream <- chat$stream_async(input$chat_user_input)
+      shinychat::chat_append("chat", stream)
+    })
+
+    shiny::observeEvent(input$close_btn, {
+      shiny::stopApp()
+    })
+  }
+
+  shiny::shinyApp(ui, server)
```
Comment (jcheng5) [inline R/shiny.R:34]: Maybe this should be `shiny::runGadget(shiny::shinyApp(ui, server))`? That would more strongly encourage RStudio to run it in the viewer pane (I think... it's been a while).

File: R/shiny.R (hunk 2 above)
Comment (jcheng5) [reply R/shiny.R:8]: In this context, Shiny is just an implementation detail 😄

File: R/shiny.R [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#' @returns A [shiny::shinyApp].
+chatApp <- function(chat) {
+  check_installed(c("bslib", "shiny", "shinychat"))
+
+  ui <- bslib::page_fluid(
+    shinychat::chat_ui("chat"),
+    shiny::actionButton(
+      "close_btn", "",
+      class = "btn-close",
+      style = "position: absolute; top: 10px; left: 10px;"
```
Comment (jcheng5) [inline R/shiny.R:16]:
```suggestion
      style = "position: absolute; top: 10px; right: 10px;"
```

Comment (hadley) [reply R/shiny.R:16]: That makes it overlap with the first chat bubble? (That is also what claude suggested 😉)

**Comment by hadley** [comment]
@jcheng5 I decided to make them both functions because that somehow feels better to me for reasons I can't fully articulate.

@topepo FWIW I've built in cancellation handling and changed `$console` from a method to a function.

**Comment by jcheng5** [comment]
I tweaked the CSS slightly, feel free to revert it.

**Review by jcheng5 — APPROVED** [review]

**Comment by hadley** [comment]
Looks much better. thanks!

**Pull request merged**